simulator/modules/core/t/unit_test.cpp

```diff
@@ -171,3 +171,23 @@ TEST_CASE( "Torture_Test: Perf_Sim, RISC-V 32 simple trace")
     CHECK( sim->run_no_limit() == Trap::HALT);
     CHECK( sim->get_exit_code() == 0);
 }
+
+TEST_CASE( "Perf_sim: Syscall flushes pipeline")
+{
+    std::istringstream input( "4\n8\n");
+    std::ostringstream output;
+    auto sim = CycleAccurateSimulator::create_simulator( "riscv32");
+    auto mem = FuncMemory::create_default_hierarchied_memory();
+    sim->set_memory( mem);
+    auto kernel = create_mars_kernel( input, output, std::cerr);
+    kernel->set_simulator( sim);
+    kernel->connect_memory( mem);
+    kernel->connect_exception_handler();
+    kernel->load_file( TEST_PATH "/rv32-scall");
+    sim->set_kernel( kernel);
+    sim->init_checker();
+    sim->set_pc( kernel->get_start_pc());
+    CHECK( sim->run_no_limit() == Trap::HALT);
+    CHECK( sim->get_exit_code() == 0);
+    CHECK( output.str().compare( "This program calculates the sum of two numbers.\nEnter 1st number: Enter 2nd number: Sum: 12\n") == 0);
```

Comment:
Actually, you do not need to check the stdout behavior, we just expect the program to pass and do not throw a CheckerMismatch